tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/FakeCtx.scala

```diff
@@ -0,0 +1,14 @@
+package at.forsyte.apalache.tla.bmcmt.stratifiedRules.aux
+
+import at.forsyte.apalache.tla.bmcmt.smt.{SolverConfig, Z3SolverContext}
+import at.forsyte.apalache.tla.lir.TlaEx
+
+/**
+ * A fake context, which just intercepts any expression added via assertGroundExpr and stores it in a Seq.
+ *
+ * We can use it in tests, to observe constraint generation without needing to instrument SMT.
+ */
+class FakeCtx extends Z3SolverContext(SolverConfig.default) {
```

Comment:
I don't have a strong preference here, but you could just use easymock's `mock`?

Reply:
We could, but that's another dependency, and I tend to avoid those if possible.

Reply:
Not sure I understand. Dependency on what? We're already using easymock.

Reply:
Hmm, I checked it out, we're actually only using it in 2 places, and only for a class with 2 unit methods, which are trivial to mock. We could just get rid of it entirely

Reply:
Erm, there's 5 tests using it. Also, I suggested that you **use** it, not that you **remove** it?

```sh
$ ag easymock
tla-typechecker/src/test/scala/at/forsyte/apalache/tla/typecheck/etc/TestConstraintSolver.scala
8:import org.scalatestplus.easymock.EasyMockSugar

tlair/src/test/scala/at/forsyte/apalache/tla/types/TestTypeUnifier.scala
8:import org.scalatestplus.easymock.EasyMockSugar

tla-typechecker/src/test/scala/at/forsyte/apalache/tla/typecheck/etc/TestTypeAliasSubstitution.scala
6:import org.scalatestplus.easymock.EasyMockSugar

tla-typechecker/src/test/scala/at/forsyte/apalache/tla/typecheck/etc/TestEtcTypeChecker.scala
7:import org.easymock.EasyMock
11:import org.scalatestplus.easymock.EasyMockSugar

passes/src/test/scala/at/forsyte/apalache/tla/typecheck/etc/TestTypeCheckerTool.scala
13:import org.easymock.EasyMock
17:import org.scalatestplus.easymock.EasyMockSugar
```

Reply:
The mock sugars don't really do anything, you can delete the extends and the tests don't fail. 

But the point is, you said that it doesn't add any value for the current tests to avoid easymock, because it's already a dependency, which is true at present. However, we can remove easymock from the codebase, and slim our dependencies, at which point having the above mock class avoids having to reintroduce easymock.